scripts/core/EXP-COMMAND-BUFER.rst

```diff
@@ -0,0 +1,116 @@
+feature will report ${X}_COMMAND_BUFFER_EXTENSION_STRING_EXP in the returned list of
+supported extensions.
+
+.. parsed-literal::
+
+    // Retrieve length of extension string
+    size_t returnedSize;
+    ${x}DeviceGetInfo(hDevice, ${X}_DEVICE_INFO_EXTENSIONS, 0, nullptr, &returnedSize);
+
+    // Retrieve extension string
+    std::unique_ptr<char[]> extensionString(new char[returnedSize]);
+    ${x}DeviceGetInfo(hDevice, ${X}_DEVICE_INFO_NAME, returnedSize, extensionString.get(), nullptr);
+    
+    std::string_view ExtensionsString(Result.get());
+    bool CmdBufferSupport = ExtensionsString.find(${X}_COMMAND_BUFFER_EXTENSION_STRING_EXP) != std::string::npos;
+
+Command-Buffer Creation
+=======================
+
+Command-Buffers are tied to a specific ${x}_context_handle_t and ${x}_device_handle_t.
+${x}CommandBufferCreateExp optionally takes a descriptor to provide additional properties
+for how the command-buffer should be constructed. There are currently no unique members
+defined for ${x}_exp_command_buffer_desc_t, however they may be added in the future.
+
+Command-buffers are reference counted and can be retained and released by calling
+${x}CommandBufferRetainExp and ${x}CommandBufferReleaseExp respectively.
+
+Appending Commands
+==================
+
+Commands can be appended to a command-buffer by calling any of the command-buffer
+append functions. Typically these closely mimic the existing enqueue functions in the
+Core API in terms of their command-specific parameters. However, they differ in that they
+take a command-buffer handle instead of a queue handle, and the dependencies and return
+parameters are sync-points instead of event handles.
+
+Currently only the following commands are supported:
+
+* ${x}CommandBufferAppendKernelLaunchExp
+* ${x}CommandBufferAppendMemcpyUSMExp
+* ${x}CommandBufferAppendMembufferCopyExp
+* ${x}CommandBufferAppendMembufferCopyRectExp
+
+It is planned to eventually support any command type from the Core API which can actually
+be appended to the equiavalent adapter native constructs.
+
+Sync-Points
+===========
+
+A sync-point is a value which represents a command inside of a command-buffer which is
+returned from command-buffer append function calls. These can be optionally passed
+to these functions to define execution dependencies on other commands within the command-buffer.
+
+Sync-points are unique and valid for use only within the command-buffer they were obtained from.
+
+.. parsed-literal::
+    // Append a memcpy with no sync-point dependencies
+    ${x}_exp_command_buffer_sync_point_t syncPoint;
+
+    ${x}CommandBufferAppendMemcpyUSMExp(hCommandBuffer, pDst, pSrc, size, 0, nullptr, &syncPoint);
+    
+    // Append a kernel launch with syncPoint as a dependency, ignore returned sync-point
+    ${x}CommandBufferAppendKernelLaunchExp(hCommandBuffer, hKernel, workDim, pGlobalWorkOffset, pGlobalWorkSize, pLocalWorkSize, 1, &syncPoint, nullptr);
```

Comment:
Formatting code blocks to shorter line lengths, e.g. 80, make them more readable in the docs.

Reply:
Most code block lines have been formatted apart from those that prevent generation, see [this comment](https://github.com/oneapi-src/unified-runtime/pull/591#discussion_r1228379992).